File: singly_linked_list/recipe-02/cxx/test_list_sort.cpp
#include "singly_linked_list.hpp"
#include <string>
#include <iostream>

void test() {
    SinglyLinkedList<std::string> words;
    words.Append("foo");
    words.Append("bar");
    words.Append("bim");
    words.Append("baz");
    words.Append("quux");

    std::cout << "The origin list" << '\n';
    for (const auto& word : words)
        std::cout << "data: " << word << '\n';

    std::cout << "Sort the list" << '\n';
    words.Sort();
    for (const auto& word : words)
        std::cout << "data: " << word << '\n';
}

int main()
{
    test();
    return 0;
}

File: singly_linked_list/recipe-02/python/test_list_sort.py
from singly_linked_list import SinglyLinkedList

def test():
    words = SinglyLinkedList()
    words.Append('foo')
    words.Append('bar')
    words.Append('bim')
    words.Append('baz')
    words.Append('quux')
    
    print("The origin list");
    for word in words:
        print("data: {}".format(word))

    print("Sort the list");
    words.Sort();
    for word in words:
        print("data: {}".format(word))

if __name__ == "__main__":
    test()
